internal/model/modelindexer/indexer.go

```diff
@@ -401,12 +477,23 @@ func (i *Indexer) runActiveIndexer() {
 		if err := active.Add(event); err != nil {
 			i.logger.Errorf("failed adding event to bulk indexer: %v", err)
 		}
-		return false
 	}
 	for !closed {
 		select {
 		case <-flushTimer.C:
+			countTimedFlush()
+		case <-idleTimer.C:
+			countTimedFlush()
 		default:
+			// When the queue utilization is below 5%, reset the idleTimer. When
+			// traffic to the APM Server is interrupted or stopped, it allows excess
+			// active indexers that have been idle for a the IdleInterval to be
+			// scaled down.
+			activeIndexers := atomic.LoadInt64(&i.activeBulkRequests)
+			lowChanCapacity := float64(len(i.bulkItems))/float64(cap(i.bulkItems)) <= 0.05
+			if lowChanCapacity && activeIndexers > 1 {
+				idleTimer.Reset(i.config.Scaling.IdleInterval)
+			}
```

Comment:
Doing this on every iteration of the loop concerns me a little bit. Would it make sense to require `FlushInterval >= IdleInterval`, and only reset the idle timer before the first iteration of the loop, and then after flushing? i.e. whenever the flush timer is inactive.

Also, I think the idle timer should only be started when autoscaling is enabled?